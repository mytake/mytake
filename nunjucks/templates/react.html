{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
