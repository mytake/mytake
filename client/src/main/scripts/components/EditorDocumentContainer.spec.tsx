import * as React from "react";
import * as renderer from "react-test-renderer";
import {
  EditorDocumentBranch,
  EditorDocumentContainerProps,
  EditorDocumentContainerState
} from "./EditorDocumentContainer";
import { documentFactLink, documentNodes } from "../utils/testUtils";

const mockFn = jest.fn();

const containerProps: EditorDocumentContainerProps = {
  active: false,
  block: {
    kind: "document",
    excerptId: "o_dRqrNJ62wzlgLilTrLxkHqGmvAS9qTpa4z4pjyFqA=",
    highlightedRange: [11, 53],
    viewRange: [0, 218]
  },
  eventHandlers: {
    onDocumentClick: mockFn
  },
  idx: 1
};

const document = {
  fact: documentFactLink.fact,
  nodes: documentNodes
};

test("Document loading", () => {
  const containerState: EditorDocumentContainerState = { loading: true };

  const tree = renderer
    .create(
      <EditorDocumentBranch
        containerProps={containerProps}
        containerState={containerState}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Successfully loaded Document", () => {
  const containerState: EditorDocumentContainerState = {
    loading: false,
    document: document
  };

  const tree = renderer
    .create(
      <EditorDocumentBranch
        containerProps={containerProps}
        containerState={containerState}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
