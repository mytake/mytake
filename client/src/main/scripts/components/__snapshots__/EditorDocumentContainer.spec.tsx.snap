// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document loading 1`] = `
<div
  className="editor__document editor__document--base editor__document--hover"
>
  <h2
    className="editor__document-title"
  >
    Loading
  </h2>
</div>
`;

exports[`Successfully loaded Document 1`] = `
<div
  className="editor__document editor__document--base editor__document--hover"
  onClick={[Function]}
>
  <div>
    <h2
      className="editor__document-title"
    >
      Amendment 13
    </h2>
    <div>
      <p
        data-char-offset={0}
      >
        Section 1. 
        <span
          className="editor__document-highlight"
        >
          Neither slavery nor involuntary servitude,
        </span>
         except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction.
      </p>
    </div>
  </div>
</div>
`;
